### PR TITLE
Chore(Docs): AWS EC2 Stress Chaos

### PR DESCRIPTION
**Pull request by avaakash**
# Harness Developer Pull Request
This PR adds the docs for AWS EC2 stress chaos experiments
1. ec2-cpu-hog
2. ec2-memory-hog
3. ec2-io-stress

In addition this PR updates
1. Diagrams for other AWS experiments - fixed ENV names
2. Removed not required pre-requisites

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [x] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [x] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/chaos-engineering/chaos-faults/aws/ec2-io-stress.md [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+type: Opaque
+stringData:
+  cloud_config.yml: |-
+    # Add the cloud AWS credentials respectively
+    [default]
+    aws_access_key_id = XXXXXXXXXXXXXXXXXXX
+    aws_secret_access_key = XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+```
+
+- If you change the secret name then please also update the `experiment.yml` ENV values for deriving the respective data from the secret. Also account for the path at which this secret is mounted as a file in the manifest ENV `AWS_SHARED_CREDENTIALS_FILE`.
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/aws/ec2-io-stress.md]:
```suggestion
## Prerequisites
:::info
- Ensure that Kubernetes Version >= 1.17
- Ensure that SSM agent is installed and running in the target EC2 instance.
- Ensure to create a Kubernetes secret having the AWS Access Key ID and Secret Access Key credentials in the `CHAOS_NAMESPACE`. A sample secret file looks like:
```yaml
apiVersion: v1
kind: Secret
metadata:
  name: cloud-secret
type: Opaque
stringData:
  cloud_config.yml: |-
    # Add the cloud AWS credentials respectively
    [default]
    aws_access_key_id = XXXXXXXXXXXXXXXXXXX
    aws_secret_access_key = XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
```
- If you change the secret name then please also update the `experiment.yml` ENV values for deriving the respective data from the secret. Also account for the path at which this secret is mounted as a file in the manifest ENV `AWS_SHARED_CREDENTIALS_FILE`.
:::
```

Comment (neelanjan00) [reply docs/chaos-engineering/chaos-faults/aws/ec2-io-stress.md]: Can we correct the format? The `info` admonition is missing

File: docs/chaos-engineering/chaos-faults/aws/ec2-memory-hog.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+type: Opaque
+stringData:
+  cloud_config.yml: |-
+    # Add the cloud AWS credentials respectively
+    [default]
+    aws_access_key_id = XXXXXXXXXXXXXXXXXXX
+    aws_secret_access_key = XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+```
+
+- If you change the secret name then please also update the `experiment.yml` ENV values for deriving the respective data from the secret. Also account for the path at which this secret is mounted as a file in the manifest ENV `AWS_SHARED_CREDENTIALS_FILE`.
```
Comment (neelanjan00) [inline docs/chaos-engineering/chaos-faults/aws/ec2-memory-hog.md]: Same comment here too, we're adding prerequisites in an `info` admonition (https://docusaurus.io/docs/next/markdown-features/admonitions), please follow that format

File: docs/chaos-engineering/chaos-faults/aws/ec2-cpu-hog.md [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+            <td> Provide the path for aws secret credentials</td>
+            <td> Defaults to <code>/tmp/cloud_config.yml</code> </td>
+        </tr>
+        <tr>
+            <td> INSTALL_DEPENDENCIES </td>
+            <td> Select to install dependencies used to run the cpu chaos. It can be either True or False</td>
+            <td> Defaults to True </td>
+        </tr>
+        <tr>
+            <td> CPU_CORE </td>
```
Comment (ispeakc0de) [inline docs/chaos-engineering/chaos-faults/aws/ec2-cpu-hog.md:113]: EC2 experiments has DOCUMENT_NAME, DOCUMENT_FORMAT, DOCUMENT_TYPE, DOCUMENT_PATH envs as well

File: docs/chaos-engineering/chaos-faults/aws/ec2-cpu-hog.md [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+```yaml
+# cpu cores to utilize
+apiVersion: litmuschaos.io/v1alpha1
+kind: ChaosEngine
+metadata:
+  name: engine-nginx
+spec:
+  engineState: "active"
+  annotationCheck: "false"
+  chaosServiceAccount: ec2-cpu-hog-sa
```
Comment (ispeakc0de) [inline docs/chaos-engineering/chaos-faults/aws/ec2-cpu-hog.md]: use the litmus-admin service account - we are not providing experiment specific rbac here

File: docs/chaos-engineering/chaos-faults/aws/ec2-io-stress.md [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+  experiments:
+  - name: ec2-io-stress
+    spec:
+      components:
+        env:
+        - name: MEMORY_CONSUMPTION
+          VALUE: '1024'
+        # ids of the ec2 instances
+        - name: EC2_INSTANCE_ID
+          value: 'instance-1'
```
Comment (ispeakc0de) [inline docs/chaos-engineering/chaos-faults/aws/ec2-io-stress.md:312]:
```suggestion
          value: 'instance-1,instance-2'
```

File: docs/chaos-engineering/chaos-faults/aws/ec2-cpu-hog.md [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+title: EC2 CPU Hog
+---
+
+## Introduction
+
+- EC2 CPU Hog contains chaos to disrupt the state of infra resources. The experiment can induce stress chaos on AWS EC2 Instance using Amazon SSM Run Command, this is carried out by using SSM Docs which is in-built in the experiment for the given chaos scenario.
+- It causes CPU Hog chaos on EC2 Instance using an SSM doc for a certain chaos duration.
+
+:::tip Fault execution flow chart
+![EC2 CPU Hog](./static/images/ec2-cpu-hog.png)
```
Comment (ispeakc0de) [inline docs/chaos-engineering/chaos-faults/aws/ec2-cpu-hog.md:12]: typo in png: REVERT TRESS --> REVERT STRESS - check other stress flow-charts as well

File: docs/chaos-engineering/chaos-faults/aws/static/manifests/ec2-cpu-hog/cpu-core-with-percentage.yaml [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# cpu core with percentage to utilize
+apiVersion: litmuschaos.io/v1alpha1
+kind: ChaosEngine
+metadata:
+  name: engine-nginx
+spec:
+  engineState: "active"
+  annotationCheck: "false"
+  chaosServiceAccount: ec2-cpu-hog-sa
```
Comment (ispeakc0de) [inline docs/chaos-engineering/chaos-faults/aws/static/manifests/ec2-cpu-hog/cpu-core-with-percentage.yaml]: chaosServiceAccount: litmus-admin -- change at all places

File: docs/chaos-engineering/chaos-faults/aws/ec2-io-stress.md (hunk 5 above)
Comment (avaakash) [reply docs/chaos-engineering/chaos-faults/aws/ec2-io-stress.md:312]: Have provided multiple values in the multiple-instances.yaml example.

**Review by ispeakc0de — APPROVED** [review]

**Pull request merged**